    //     throw;  // rethrow current exception
// no 'finally' block!